packages/backend/src/services/PersonalAccessTokenService.ts

```diff
@@ -58,4 +60,42 @@ export class PersonalAccessTokenService extends BaseService {
     ): Promise<PersonalAccessToken[]> {
         return this.personalAccessTokenModel.getAllForUser(user.userId);
     }
+
+    async rotatePersonalAccessToken(
+        user: Pick<SessionUser, 'userUuid'>,
+        personalAccessTokenUuid: string,
+        data: { expiresAt: Date },
+    ): Promise<PersonalAccessTokenWithToken> {
+        if (data.expiresAt.getTime() < Date.now()) {
+            throw new ParameterError('Expire time must be in the future');
+        }
+
+        const existingToken = await this.personalAccessTokenModel.getUserToken({
+            userUuid: user.userUuid,
+            tokenUuid: personalAccessTokenUuid,
+        });
+
+        if (!existingToken.expiresAt) {
```

Comment:
since this is a business decision, and not a security/tecnical issue, I 'd add a comment here, in case someone has a use case that forces us to remove it. So we don't need to remember why we do this in the first place in a few months.